modin/experimental/xgboost/xgboost_ray.py

```diff
@@ -297,44 +295,20 @@ def _get_num_actors(num_actors):
     """
     min_cpus_per_node = _get_min_cpus_per_node()
     if num_actors == "default_train":
-        num_actors_per_node = max(1, int(min_cpus_per_node // 2))
+        num_actors_per_node = max(1, int(min_cpus_per_node // 16))
         return num_actors_per_node * len(ray.nodes())
     elif num_actors == "default_predict":
-        num_actors_per_node = max(1, int(min_cpus_per_node // 8))
+        num_actors_per_node = max(1, int(min_cpus_per_node // 16))
         return num_actors_per_node * len(ray.nodes())
     elif isinstance(num_actors, int):
+        assert (
+            num_actors % len(ray.nodes()) == 0
+        ), "`num_actors` must be a multiple to number of nodes in Rau cluster."
         return num_actors
     else:
-        RuntimeError("`num_actors` must be int or None")
-
-
-def _create_placement_group(num_cpus_per_actor, num_actors):
-    """
-    Create Ray placement group to grab resources.
-
-    Parameters
-    ----------
-    num_cpus_per_actor : int
-        Number of CPUs per actor.
-    num_actors : int
-        Number of actors.
-
-    Returns
-    -------
-    ray.util.PlacementGroup
-        Placement group with grabbed resources.
-    """
-    cpu_bundle = {"CPU": num_cpus_per_actor}
-    bundles = [cpu_bundle for _ in range(num_actors)]
-
-    pg = placement_group(bundles, strategy="SPREAD")
-
-    ready, _ = ray.wait([pg.ready()], timeout=100)
-
-    if ready is None:
-        raise TimeoutError("Placement group creation timeout.")
-
-    return pg
+        raise RuntimeError(
```

Comment:
It seems to me the PR contains the changes unrelated to the issue. Can you leave the changes required to resolve the issue and create an issue to refactor?

Reply:
The PR contains only 2 changes unrelated to the issue:
1. Changing of name `max_node_cpus -> min_node_cpus`.
2. Adding `raise` before RuntimeError.

I'm not sure that we should add this minor changes in separate PR, because those changes don't include a lot of semantic load on this PR.

Reply:
I still feel these changes should not belong to this PR. A user might pass a value of `num_actors` that will lead to this branch and the user won't get `ValueError`. This looks like fixing a different issue. So I suggest creating an issue to refactor that will contain all the changes not related to this issue and something from #2936.

Reply:
Ok, corresponded issue #3304 was created. Only changes described in https://github.com/modin-project/modin/pull/3220#discussion_r680732979 are relevant from list described in #2936.

Reply:
Thanks! This one ([link](https://github.com/modin-project/modin/pull/2936#discussion_r611422935)) might be relevant.

Reply:
In [link](https://github.com/modin-project/modin/pull/2936#discussion_r611422935) `parts_ips_ref` is tuple and it should be converted to list